src/managers/config.py

```diff
@@ -465,22 +507,13 @@ def set_environment(self) -> None:
             self.log_level,
         ]
 
-        def map_env(env: list[str]) -> dict[str, str]:
-            map_env = {}
-            for var in env:
-                key = "".join(var.split("=", maxsplit=1)[0])
-                value = "".join(var.split("=", maxsplit=1)[1:])
-                if key:
-                    # only check for keys, as we can have an empty value for a variable
-                    map_env[key] = value
-            return map_env
-
         raw_current_env = self.workload.read("/etc/environment")
         current_env = map_env(raw_current_env)
 
         updated_env = current_env | map_env(updated_env_list)
         content = "\n".join([f"{key}={value}" for key, value in updated_env.items()])
         self.workload.write(content=content, path="/etc/environment")
+        self.workload.write(content=content, path=f'{BROKER.paths["CONF"]}/.env')
```

Comment:
**question:** I know we talked about this, but what was the reasoning behind having two environments with the same content here?

Reply:
So, the issue at first was that I was using a Jaas config file to pass credentials. Since the binary either looks in a hardcoded path or through the `KAFKA_OPTS` env var, I wanted a way to have different env vars depending on the workload. At first, I was just overwriting `/etc/environment` before every command, and wanted an easy way to access those vars when SSh-ing into the machine. Hence, the .env files.

Then, I thought that I might as well avoid `/etc/environment` altogether and figure out a way to use those specific .env. This is doable any time we use `subprocess`functions ourselves, but not when we use stuff from `operator_libs_linux`, so we were still stuck with overwriting `/etc/environment`.

In the end, we do not use a Jaas file any more, so the var issue is gone too. I can get rid of the .env files, but I wanted to check if there is still a use for them

Reply:
I think long term, we should make a PR to `operator_libs_linux.v1.snap` lib upstream to support passing env, and then the `.env` file idea is MUCH better than a polluted `/etc/environment`, but let's do that at a later date when we have some spare time.

Reply:
.envs removed in  ad5961c16b72616404a171ddf114ada5c8c45e5f